// @aws-sdk/client-identitystore/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListUsersResponse = exports.User = exports.ListUsersRequest = exports.ListGroupsResponse = exports.Group = exports.ListGroupsRequest = exports.Filter = exports.DescribeUserResponse = exports.DescribeUserRequest = exports.ValidationException = exports.ThrottlingException = exports.ResourceNotFoundException = exports.ResourceType = exports.InternalServerException = exports.DescribeGroupResponse = exports.DescribeGroupRequest = exports.AccessDeniedException = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const IdentitystoreServiceException_1 = require("./IdentitystoreServiceException");
class AccessDeniedException extends IdentitystoreServiceException_1.IdentitystoreServiceException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
        this.RequestId = opts.RequestId;
    }
}
exports.AccessDeniedException = AccessDeniedException;
var DescribeGroupRequest;
(function (DescribeGroupRequest) {
    DescribeGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeGroupRequest = exports.DescribeGroupRequest || (exports.DescribeGroupRequest = {}));
var DescribeGroupResponse;
(function (DescribeGroupResponse) {
    DescribeGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeGroupResponse = exports.DescribeGroupResponse || (exports.DescribeGroupResponse = {}));
class InternalServerException extends IdentitystoreServiceException_1.IdentitystoreServiceException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
        this.RequestId = opts.RequestId;
    }
}
exports.InternalServerException = InternalServerException;
var ResourceType;
(function (ResourceType) {
    ResourceType["GROUP"] = "GROUP";
    ResourceType["IDENTITY_STORE"] = "IDENTITY_STORE";
    ResourceType["USER"] = "USER";
})(ResourceType = exports.ResourceType || (exports.ResourceType = {}));
class ResourceNotFoundException extends IdentitystoreServiceException_1.IdentitystoreServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.ResourceType = opts.ResourceType;
        this.ResourceId = opts.ResourceId;
        this.Message = opts.Message;
        this.RequestId = opts.RequestId;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class ThrottlingException extends IdentitystoreServiceException_1.IdentitystoreServiceException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
        this.RequestId = opts.RequestId;
    }
}
exports.ThrottlingException = ThrottlingException;
class ValidationException extends IdentitystoreServiceException_1.IdentitystoreServiceException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
        this.RequestId = opts.RequestId;
    }
}
exports.ValidationException = ValidationException;
var DescribeUserRequest;
(function (DescribeUserRequest) {
    DescribeUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeUserRequest = exports.DescribeUserRequest || (exports.DescribeUserRequest = {}));
var DescribeUserResponse;
(function (DescribeUserResponse) {
    DescribeUserResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.UserName && { UserName: smithy_client_1.SENSITIVE_STRING }),
    });
})(DescribeUserResponse = exports.DescribeUserResponse || (exports.DescribeUserResponse = {}));
var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.AttributeValue && { AttributeValue: smithy_client_1.SENSITIVE_STRING }),
    });
})(Filter = exports.Filter || (exports.Filter = {}));
var ListGroupsRequest;
(function (ListGroupsRequest) {
    ListGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Filters && { Filters: obj.Filters.map((item) => Filter.filterSensitiveLog(item)) }),
    });
})(ListGroupsRequest = exports.ListGroupsRequest || (exports.ListGroupsRequest = {}));
var Group;
(function (Group) {
    Group.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Group = exports.Group || (exports.Group = {}));
var ListGroupsResponse;
(function (ListGroupsResponse) {
    ListGroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGroupsResponse = exports.ListGroupsResponse || (exports.ListGroupsResponse = {}));
var ListUsersRequest;
(function (ListUsersRequest) {
    ListUsersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Filters && { Filters: obj.Filters.map((item) => Filter.filterSensitiveLog(item)) }),
    });
})(ListUsersRequest = exports.ListUsersRequest || (exports.ListUsersRequest = {}));
var User;
(function (User) {
    User.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.UserName && { UserName: smithy_client_1.SENSITIVE_STRING }),
    });
})(User = exports.User || (exports.User = {}));
var ListUsersResponse;
(function (ListUsersResponse) {
    ListUsersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Users && { Users: obj.Users.map((item) => User.filterSensitiveLog(item)) }),
    });
})(ListUsersResponse = exports.ListUsersResponse || (exports.ListUsersResponse = {}));
